File: CP01007.py
t = int(input())
while(t):
    n = int(input())
    a=list(map(lambda x:int(x),input().strip().split(' ')))
    res=[]
    res.append(a[n-1])
    for i in range(0,n):
        for j in range(i+1,n):
            if(a[j]>=a[i]): 
                break
            if(j==n-1):
                res.append(a[i])
    res.sort(reverse=True)
    for i in res:
        print(i,end=' ')
    print()
    t-=1

File: CP01007.cpp
#include <iostream>
#include<vector>
#include<algorithm>
using namespace std;


int main()
{
    int t=0;
    cin>>t;
    while(t--){
        int n;
        cin >>n;
        int *a = new int[n]{0};
        vector<int> res;
        for(int i=0;i<n;i++){
            cin>>a[i];
        }
        res.push_back(a[n-1]);
        for(int i=n-2;i>=1;i--){
           for(int j=i+1;j<n;j++){
               if(a[j]>=a[i]){
                   break;
               }
               if(j==n-1){
                   res.push_back(a[i]);
               }
           }
        }
        sort(res.begin(),res.end());
        for (int i = res.size()-1; i >=0; i--)
        {
            cout<<res[i]<<' ';
        }
        cout<<endl;
    }
}